lib/matplotlib/patches.py

```diff
@@ -2166,22 +2166,23 @@ def draw_bbox(bbox, renderer, color='k', trans=None):
     r.draw(renderer)
 
 
-def _simpleprint_styles(_styles):
-    """
-    A helper function for the _Style class.  Given the dictionary of
-    {stylename: styleclass}, return a string rep of the list of keys.
-    Used to update the documentation.
-    """
-    return "[{}]".format("|".join(map(" '{}' ".format, _styles)))
-
-
 class _Style:
     """
     A base class for the Styles. It is meant to be a container class,
     where actual styles are declared as subclass of it, and it
     provides some helper functions.
     """
 
+    def __init_subclass__(cls):
+        _docstring.interpd.update({
```

Comment:
```suggestion
        # automatically perform docstring interpolation on the subclasses
        # This allows listing the supported styles via
        # - %(BoxStyle:table)s
        # - %(ConnectionStyle:table)s
        # - %(ArrowStyle:table)s
        # and additionally adding .. ACCEPTS: blocks via
        # - %(BoxStyle:table_and_accepts)s
        # - %(ConnectionStyle:table_and_accepts)s
        # - %(ArrowStyle:table_and_accepts)s
        _docstring.interpd.update({
```
When encountering the above placeholders in a docstring, it's extremely difficult to find out where they are resolved because you can't add a local comment in the docstring (well, you can in rst but I bet most people would not recognize it as comment and only be more confused). Also the identifiers "BoxStyle:table" do not exist anywhere in the code because the are auto-generated via an f-string.
This explicit listing is my clumsy attempt to give people a chance to find something when they do a full text search on e.g. "BoxStyle:table" .

Reply:
Sure, looks reasonable.